### PR TITLE
python virtualenv per project and correct path for embedded interpreter

### PR DESCRIPTION
**Pull request by dydek**

### CODE REVIEW

File: backend/aiconsole/core/code_running/code_interpreters/languages/python.py [hunk 1]
```diff
@@ -100,14 +102,14 @@ def preprocess_python(code: str, materials: List[Material]):
...
 import traceback
-from aiconsole.dev.credentials import MissingCredentialException
-
-try:
-{newline.join(("    " + line) for line in code.split(newline) if line.strip())}
-except MissingCredentialException as e:
-    print(e)
-except Exception:
-    traceback.print_exc()
+#from aiconsole.dev.credentials import MissingCredentialException
```
Comment (dydek) [inline backend/aiconsole/core/code_running/code_interpreters/languages/python.py:105]: this is something I'm not so sure about - if we want to still use it we'd need to install aiconsole package in every venv - which is possible but I wanted to have your opinion before that

File: backend/aiconsole/core/code_running/code_interpreters/subprocess_code_interpreter.py [hunk 2]
```diff
@@ -82,15 +85,16 @@ def start_process(self):
                                         stdin=subprocess.PIPE,
                                         stdout=subprocess.PIPE,
                                         stderr=subprocess.PIPE,
+                                        env=self._prepare_env(),
```
Comment (dydek) [inline backend/aiconsole/core/code_running/code_interpreters/subprocess_code_interpreter.py:88]: the main part to modify the env for the execution

File: backend/aiconsole/core/code_running/code_interpreters/languages/python.py [hunk 3]
```diff
@@ -43,13 +43,15 @@
...
+
 class Python(SubprocessCodeInterpreter):
     file_extension = "py"
     proper_name = "Python"
 
     def __init__(self):
         super().__init__()
-        self.start_cmd = sys.executable + " -i -q -u"
+        # the python command itself is provided by the virtual environment
+        self.start_cmd = "python -i -q -u"
```
Comment (dydek) [inline backend/aiconsole/core/code_running/code_interpreters/languages/python.py:54]: probably ( I guess ) for win32 it should be the name with `exe` -  a small fix is required probably

File: backend/aiconsole/core/code_running/code_interpreters/languages/python.py (hunk 1 above)
Comment (cielecki) [reply backend/aiconsole/core/code_running/code_interpreters/languages/python.py:105]: yes we need to do that - some of the materials need to have access to the packages.

There is a space to maybe seperate a package for materials development that will be smaller but I'm not sure if it's worth the effort.

**Review by cielecki — CHANGES_REQUESTED** [review]

**Pull request merged**